api/errdefs/helpers.go

```diff
@@ -0,0 +1,240 @@
+package errdefs
+
+import "context"
+
+type errNotFound struct{ error }
+
+func (errNotFound) NotFound() {}
+
+func (e errNotFound) Cause() error {
+	return e.error
+}
```

Comment:
This is the opposite of `containerd`, which uses instances of an error as the root cause, and then wraps with more context. That's probably ok, but something we should be careful of when converting `containerd` errors. I guess this way requires `getImplementer()`, since `errors.Cause()` will not return the correct error (another thing we need to be watch out for).

Reply:
errdefs here is implemented slightly differently from containerd, where containerd has a set of base errors like `ErrNotFound = errors.New("not found")` which you wrap and then compare against the base error like `if errors.Cause(err) == ErrNotFound`.
But yeah we need to be careful any time we are dealing with errors from other repos.